pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/utilities/ColumnDescriptor.java

```diff
@@ -65,9 +65,13 @@ public ColumnDescriptor(ColumnDescriptor copy) {
         this.dbColumnName = copy.dbColumnName;
         this.dbColumnIndex = copy.dbColumnIndex;
         this.dbColumnTypeName = copy.dbColumnTypeName;
-        System.arraycopy(this.dbColumnTypeModifiers, 0,
-                copy.dbColumnTypeModifiers, 0,
-                this.dbColumnTypeModifiers.length);
+        if (copy.dbColumnTypeModifiers != null
+                && copy.dbColumnTypeModifiers.length > 0) {
+            this.dbColumnTypeModifiers = new Integer[copy.dbColumnTypeModifiers.length];
```

Comment:
Just a cosmetic enhancement to use Arrays.copyOf instead which implicitly uses System.arraycopy and doesn't require explicit object intialization